src/lib.rs

```diff
@@ -47,6 +45,8 @@ impl Configuration {
         cache_expiration: Option<String>,
         default_quality: Option<String>,
     ) -> Self {
+        // I see two collect calls in one chain and that sets off alarms on efficiency.
+        // Since this is not a critical part of the code for optimization, I wouldn't worry.
```

Comment:
What is the performance implication of running `collect` multiple times? How would you structure this to avoid two `collect` calls?

Reply:
In this case, you are creating a char iterator from a String, filtering out whitespace, and then collecting into a new String - that will lead to a new allocation for the new String to contain the remaining characters from the iterator and filter.
Then you split and map back each split piece to a String - this portion may be just fine since the Split iterator can still be efficient and not allocate new Strings for the pieces of the split. Then you collect again to get a Vec<String> - this will also cause an allocation for the Vec and move each String into it (probably allocating for each String as well).